spec/controllers/users/omniauth_callbacks_controller_spec.rb

```diff
@@ -0,0 +1,170 @@
+describe Users::OmniauthCallbacksController do
+  
+  def mock_auth_hash
+    {
+      'provider':'google_oauth2',
+      'uid': '12345',
+      'info': {
+        'name':'Test user',
+        'email':'testgoogle@example.com'
+      }
+    }
+  end
+
+  def create_invalid_mock_hash(opts={})
+    new_hash = mock_auth_hash.merge(opts)
+    OmniAuth.config.add_mock(:google_oauth2, new_hash)
+    request.env['omniauth.auth'] = OmniAuth.config.mock_auth[:google_oauth2]
+  end
+
+  before(:each) do
+    
+    request.env["devise.mapping"] = Devise.mappings[:user]
```

Comment:
we need to explicitly tell Devise which mapping to use because in testing
we are bypassing the router